rewrite-javascript/src/main/java/org/openrewrite/javascript/rpc/JavaScriptRewriteRpc.java

```diff
@@ -112,16 +104,189 @@ public int installRecipes(String packageName, @Nullable String version) {
+                                            "  \"dependencies\": {\n" +
+                                            "    \"@openrewrite/rewrite\": \"next\",\n" +
+                                            "    \"typescript\": \"latest\"\n" +
+                                            "  }\n" +
+                                            "}\n";
+
+                Path packageJsonPath = tempDir.resolve("package.json");
+                Files.write(packageJsonPath, packageJsonContent.getBytes(StandardCharsets.UTF_8));
+
+                Path errorFile = Files.createTempFile("npm-install-error", ".log");
+
+                try {
+                    // Run npm install to download dependencies
+                    ProcessBuilder npmInstall = new ProcessBuilder("npm", "install")
+                            .directory(tempDir.toFile())
+                            .redirectOutput(ProcessBuilder.Redirect.to(new File("/dev/null")))
+                            .redirectError(ProcessBuilder.Redirect.to(errorFile.toFile()));
+
+                    Process process = npmInstall.start();
+                    int exitCode = process.waitFor();
+
+                    if (exitCode != 0) {
+                        // Read error content from the temporary file
+                        String errorContent = "";
+                        try {
+                            errorContent = new String(Files.readAllBytes(errorFile), StandardCharsets.UTF_8);
+                        } catch (IOException e) {
+                            errorContent = "Unable to read error output: " + e.getMessage();
+                        }
+
+                        throw new RuntimeException("Failed to install npm dependencies in temporary directory. Exit code: " +
+                                                   exitCode + ". Error output: " + errorContent);
+                    }
+                } finally {
+                    try {
+                        Files.deleteIfExists(errorFile);
+                    } catch (IOException ignore) {
+                    }
+                }
+
+                return tempDir.resolve("node_modules/@openrewrite/rewrite/dist");
+            } catch (IOException | InterruptedException e) {
+                throw new RuntimeException("Failed to initialize default installation directory", e);
+            }
+        }
+
+        public Builder marketplace(Environment marketplace) {
+            this.marketplace = marketplace;
+            return this;
+        }
+
+        public Builder nodePath(Path nodePath) {
+            this.nodePath = nodePath;
+            return this;
+        }
+
+        public Builder installationDirectory(Path installationDirectory) {
+            this.installationDirectory = installationDirectory;
+            return this;
+        }
+
+        public Builder socket(int port) {
+            this.port = port;
+            return this;
+        }
+
+        public Builder trace(boolean trace) {
```

Comment:
Change this to `MessageHandler` instead, which would flow down to an argument on `JsonRpc` as a substitute for the `trace` boolean flag. In addition to `TraceMessageHandler`, we may want to choose an alternative `MessageHandler` to change the over-the-wire serialization format to msgpack, etc.